src/transforms/mergeExtensions.ts

```diff
@@ -14,8 +20,13 @@ export function mergeExtensions(doc: DocumentNode): DocumentNode {
       if (t.directives != null || t.interfaces != null) {
         throw new Error("Unexpected directives or interfaces on Extension");
       }
-      fields.extend(t.name.value, t.fields);
-      return null;
+      const nameDef = ctx.getNameDefinition(t.name);
+      if (nameDef && nameDef.externalImportPath) {
+        return t;
```

Comment:
We should check the following transforms. I think they may implictly assume there are no more extensions. For example: `customSpecValidations`

Reply:
It should work fine, as custom spec validations only check that the types have fields and this should be the case even with extensions. We only don't merge extensions for external types.

Reply:
I've added test to make sure.